xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioDSP.h

```diff
@@ -0,0 +1,1288 @@
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+
+    ///
+    /// @brief DSP post processing
+    /// On the post processing can be things performed with additional channel
+    /// upmix like 6.1 to 7.1
+    /// or frequency/volume corrections, speaker distance handling, equalizer... .
+    /// All DSP add-ons allowed to-do this.
+    /// @param handle identification data for stream
+    /// @param mode_id The mode inside add-on which must be performed on call. Id
+    /// is set from add-on by iModeNumber on AE_DSP_MODE structure during
+    /// RegisterMode callback, and can be defined from add-on as a structure
+    /// pointer or anything else what is needed to find it.
+    /// @param array_in Pointer to input data memory
+    /// @param array_out Pointer to output data memory
+    /// @param samples Amount of samples inside array_in
+    /// @return Amount of samples processed
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual unsigned int PostProcess(const ADDON_HANDLE handle, unsigned int mode_id, float **array_in, float **array_out, unsigned int samples) { return 0; }
+
+    //--------------------------------------------------------------------------
+    //@}
+
+    /** @name DSP Post re-sampling
+    *  @remarks Only used by KODI if bSupportsOutputResample is set to true.
+    */
+    //@{
+    //==========================================================================
+    ///
+    /// @brief If the add-on operate with buffered arrays and the output size
+    /// can be higher as the input
+    /// it becomes asked about needed size before any OutputResampleProcess call.
+    /// @param handle identification data for stream
+    /// @return The needed size of output array or 0 if no changes within it
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual unsigned int OutputResampleProcessNeededSamplesize(const ADDON_HANDLE handle) { return 0; }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// @brief Re-sampling after master processing becomes performed with it if
+    /// needed, only
+    /// one add-on can perform it.
+    /// @param handle identification data for stream
+    /// @param array_in Pointer to input data memory
+    /// @param array_out Pointer to output data memory
+    /// @param samples Amount of samples inside array_in
+    /// @return Amount of samples processed
+    /// @remarks Optional. Is set by AE_DSP_ADDON_CAPABILITIES and asked with
+    /// GetCapabilities
+    ///
+    virtual unsigned int OutputResampleProcess(const ADDON_HANDLE handle, float **array_in, float **array_out, unsigned int samples) { return 0; }
```

Comment:
@ksooo how exact you mean with "refs"?

You mean to use as `const float *array_in[], float *array_out[]`, with the `[]`?